### PR TITLE
sql: add ImportType to descriptor

### PR DESCRIPTION
**Pull request by stevendanna**
This adds an ImportType field to table descriptors. We need this field
if we want to support disabling ImportEpoch writing via a cluster
setting since we won't necessarily be able to trust that a non-zero
ImportEpoch is actually from the currently in-progress IMPORT.

Epic: none
Release note: None

### CODE REVIEW

**Comment by stevendanna** [comment]
First commit is #120407

File: pkg/sql/catalog/descpb/structured.proto [hunk 1]
```diff
@@ -1210,7 +1210,28 @@ message TableDescriptor {
   // gets incremented while preparing the table for ingestion.
   optional uint32 import_epoch = 59 [(gogoproto.nullable) = false, (gogoproto.customname) = "ImportEpoch"];
 
-  // Next ID: 60
+  // ImportType is the type of import that is running if
+  // ImportStartWallTime is set.
+  optional ImportType import_type = 60 [(gogoproto.nullable) = false, (gogoproto.customname) = "ImportType"];
```
Comment (msbutler) [inline pkg/sql/catalog/descpb/structured.proto:1215]: could this be a bool instead with the var name `epoch_based_import_runnning`? This feels cleaner then import_type==0  (i.e. `IMPORT_WITH_START_TIME_ONLY`)  implying no import running or non-epoch based import.

**Comment by msbutler** [comment]
it may also be worth adding a small unit test for this

File: pkg/sql/catalog/descpb/structured.proto (hunk 1 above)
Comment (stevendanna) [reply pkg/sql/catalog/descpb/structured.proto:1215]: I went back and forth on this. At the time it seemed likely that we might end up with a few more types if we eventually want to differentiate between empty-vs-non-empty as well like you had in your original PR. If so, I'd slightly prefer an Enum or a uint64 with different bits assigned to different properties of the import.

But, I can change it to a bool.

Comment (msbutler) [reply pkg/sql/catalog/descpb/structured.proto:1215]: fine by me!

**Review by msbutler — APPROVED** [review]
thanks for adding that test case!

**Comment by msbutler** [comment]
closing in favor of #121042

**Comment by msbutler** [comment]
Changed my mind. I'm reviving this given [my comment ](https://github.com/cockroachdb/cockroach/pull/121042#issuecomment-2019351694)

**Review by dt — APPROVED** [review]

**Review by fqazi — APPROVED** [review]
:lgtm:

> Reviewed 4 of 6 files at r2, 1 of 10 files at r3, 7 of 7 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/120809#-:-Nu-7qW-8VFglrw8n2aC:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @mgartner and @stevendanna)

**Comment by dt** [comment]
bors r=dt

**Pull request merged**